pkg/build/api/v1beta1/conversion.go

```diff
@@ -164,7 +176,11 @@ func init() {
 		},
 		func(in *DockerBuildStrategy, out *newer.DockerBuildStrategy, s conversion.Scope) error {
 			out.NoCache = in.NoCache
-			out.PullSecretName = in.PullSecretName
+			if in.PullSecret != nil {
```

Comment:
These types seem to match exactly.  Why do we have conversions for them?

If they do they require explicit handling, will the default convert work correctly for most of the conversion? See https://github.com/openshift/origin/blob/master/pkg/authorization/api/v1beta1/conversion.go#L16 .